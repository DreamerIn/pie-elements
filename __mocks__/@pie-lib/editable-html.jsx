const EditableHtml = () => <div>EditableHtml</div>;

export default EditableHtml;
