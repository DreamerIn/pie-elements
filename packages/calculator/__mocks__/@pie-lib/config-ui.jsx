const config = {
  InputRadio: () => <div>Input Radio</div>,
  TwoChoice: () => <div>Two Choice</div>
}

module.exports = config;