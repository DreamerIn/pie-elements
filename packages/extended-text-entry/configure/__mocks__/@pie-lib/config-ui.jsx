import React from 'react';
module.exports = {
  NumberTextField : () => <div>text</div>
};