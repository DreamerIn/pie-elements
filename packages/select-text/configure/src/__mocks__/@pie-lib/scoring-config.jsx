import React from 'react';
module.exports = () => <div>ScoringConfig</div>;
