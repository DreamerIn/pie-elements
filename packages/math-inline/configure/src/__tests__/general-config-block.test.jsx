import * as React from 'react';
import GeneralConfigBlock from '../general-config-block';
import Button from '@material-ui/core/Button';
import Input from '@material-ui/core/Input';
import { InputContainer } from '@pie-lib/config-ui';
import { FeedbackConfig } from '@pie-lib/config-ui';
import { shallowChild } from '@pie-lib/test-utils';
import { shallow } from 'enzyme';
import cloneDeep from 'lodash/cloneDeep';

jest.mock('../answer-block', () => () => <div>AnswerBlock</div>);

jest.mock('@pie-lib/math-toolbar', () => () => <div>MathToolbar</div>);

const defaultProps = {
  id: '1',
  element: 'math-inline',
  feedback: {
    correct: {
      type: 'none',
      default: 'Correct'
    },
    partial: {
      type: 'none',
      default: 'Nearly'
    },
    incorrect: {
      type: 'none',
      default: 'Incorrect'
    }
  }
};

describe('GeneralConfigBlock', () => {
  let wrapper;
  let props;
  let component;

  beforeEach(() => {
    props = {
      model: defaultProps,
      onResponseTypeChange: jest.fn(),
      onLayoutChange: jest.fn(),
      onChange: jest.fn()
    };

    wrapper = shallow(<GeneralConfigBlock {...props} />);
  });

  it('renders correctly', () => {
    component = wrapper.instance();

    expect(wrapper).toMatchSnapshot();
  });
});
