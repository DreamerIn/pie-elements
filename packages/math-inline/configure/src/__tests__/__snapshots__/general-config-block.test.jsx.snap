// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneralConfigBlock renders correctly 1`] = `
<GeneralConfigBlock
  classes={
    Object {
      "container": "GeneralConfigBlock-container-1",
      "input": "GeneralConfigBlock-input-2",
      "inputContainer": "GeneralConfigBlock-inputContainer-5",
      "responseEditor": "GeneralConfigBlock-responseEditor-6",
      "selectContainer": "GeneralConfigBlock-selectContainer-4",
      "templateTitle": "GeneralConfigBlock-templateTitle-3",
    }
  }
  model={
    Object {
      "element": "math-inline",
      "feedback": Object {
        "correct": Object {
          "default": "Correct",
          "type": "none",
        },
        "incorrect": Object {
          "default": "Incorrect",
          "type": "none",
        },
        "partial": Object {
          "default": "Nearly",
          "type": "none",
        },
      },
      "id": "1",
    }
  }
  onChange={[MockFunction]}
  onLayoutChange={[MockFunction]}
  onResponseTypeChange={[MockFunction]}
/>
`;
